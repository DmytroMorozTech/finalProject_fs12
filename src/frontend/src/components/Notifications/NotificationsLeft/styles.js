import { makeStyles } from '@material-ui/core/styles'

export default makeStyles((theme) => ({

  notificationsLeft: {
    border: theme.border.simple,
    borderRadius: theme.shape.medium,
    backgroundColor: theme.palette.common.white,
    paddingTop: theme.spacing(2)
  },

  notifications: {
    padding: theme.spacing(2)
  },

  title: {
    margin: theme.spacing(2),
    color: theme.palette.grey[900]
  },

  description: {
    fontWeight: theme.typography.fontWeightRegular,
    color: theme.palette.grey[500],
    lineHeight: theme.typography.h3.lineHeight,
    marginBottom: theme.spacing(2)
  },

  line: {
    height: '1px',
    border: '0',
    width: '100%',
    margin: 0,
    backgroundColor: theme.palette.grey[300]
  },

  settings: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    backgroundColor: theme.palette.grey[100],
    paddingTop: theme.spacing(4),
    paddingBottom: theme.spacing(4),
    borderRadius: theme.shape.down
  },

  link: {
    textDecoration: 'none',
    color: theme.palette.primary.main,
    fontWeight: theme.typography.fontWeightBold,
    fontSize: theme.typography.h5.fontSize,
    cursor: 'pointer'
  }
}))
