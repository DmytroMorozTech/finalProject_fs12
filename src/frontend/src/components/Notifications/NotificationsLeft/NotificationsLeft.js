import styles from './styles'
import Typography from '@material-ui/core/Typography'
import SharedButton from '../../../shared/SharedButton/SharedButton'
import { markAllNotificationsAsViewedAction } from '../../../redux/Notification/notificationActions'
import { useDispatch, useSelector } from 'react-redux'
import { numberOfNotificationsSelector } from '../../../redux/Notification/notificationSelector'
import React from 'react'

function NotificationsLeft () {
  const classes = styles()
  const dispatch = useDispatch()
  const numberOfNotifications = useSelector(numberOfNotificationsSelector)

  return (
    <div className={classes.notificationsLeft}>
      <div className={classes.notifications}>
        <Typography variant="h5" align="center" className={classes.title}>
          Notifications
        </Typography>
        <Typography variant="h5" align="center" className={classes.description}>
          Youâ€™re all caught up! Check back later for new notifications
        </Typography>
      </div>
      <hr className={classes.line}/>
      <div className={classes.read}>
        <SharedButton
          disabled={numberOfNotifications === 0}
          onClick={() => dispatch(markAllNotificationsAsViewedAction())}
          title='Mark all as viewed'
        />
      </div>
    </div>
  )
}

export default NotificationsLeft
