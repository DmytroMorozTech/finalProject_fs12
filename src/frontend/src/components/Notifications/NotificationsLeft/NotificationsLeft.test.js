import { shallow } from 'enzyme'
import React from 'react'
import NotificationsLeft from './NotificationsLeft'
import * as redux from 'react-redux'

jest.mock('./styles', () => () => ({}))

const setUp = () => shallow(<NotificationsLeft />)

describe('should render NotificationsLeft component', () => {
  let spyOnUseSelector
  let spyOnUseDispatch
  let mockDispatch

  let component
  beforeEach(() => {
    spyOnUseSelector = jest.spyOn(redux, 'useSelector')
    spyOnUseSelector.mockReturnValue(false)
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    component = setUp()
  })

  afterEach(() => {
    jest.restoreAllMocks()
  })

  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  it('should work markAllNotificationsAsViewedAction dispatch', () => {
    expect(mockDispatch).toHaveBeenCalledTimes(0)
    component.find('SharedButton').simulate('click')
    expect(mockDispatch).toHaveBeenCalledTimes(1)
  })
})
