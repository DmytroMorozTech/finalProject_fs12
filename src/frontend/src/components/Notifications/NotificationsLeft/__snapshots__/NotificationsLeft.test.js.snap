// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NotificationsLeft component to match snapshot 1`] = `
<div>
  <div>
    <WithStyles(ForwardRef(Typography))
      align="center"
      variant="h5"
    >
      Notifications
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      align="center"
      variant="h5"
    >
      Youâ€™re all caught up! Check back later for new notifications
    </WithStyles(ForwardRef(Typography))>
  </div>
  <hr />
  <div>
    <SharedButton
      disabled={false}
      onClick={[Function]}
      title="Mark all as viewed"
    />
  </div>
</div>
`;
